dpnp/backend_mathematical.pyx

```diff
@@ -201,6 +202,40 @@ cpdef dparray dpnp_divide(dparray x1, dparray x2):
     return call_fptr_2in_1out(DPNP_FN_DIVIDE, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_ediff1d(dparray x1, dparray to_end, dparray to_begin):
+
+    types_map = {
+        'int32': dpnp.int64,
+        'int64': dpnp.int64,
+        'float32': dpnp.float32,
+        'float64': dpnp.float64
+    }
+
+    res_type = types_map[x1.dtype.name]
```

Comment:
```suggestion
    types_map = {
        dpnp.int32: dpnp.int64,
        dpnp.int64: dpnp.int64,
        dpnp.float32: dpnp.float32,
        dpnp.float64: dpnp.float64
    }

    res_type = types_map[x1.dtype]
```
In this case no need to compare strings - pointers will be compared.
Also, what will happen if `dpnp.complex128` come into function?

Reply:
Looks like `KeyError: <dpnp.complex128>` will be raised in case of `x1.dtype == dpnp.complex128`. Maybe better to raise `TypeError` with `checker_throw_type_error(...)` in this case.

Reply:
or change this code in the way to use existed mechanism from backend. It might be a step 2. 
As a first step - it is better to avoid string comparison as much as possible.